interp/interp.go

```diff
@@ -670,10 +670,17 @@ func (r *Runner) redir(rd *syntax.Redirect) (io.Closer, error) {
 	case syntax.RdrOut, syntax.RdrAll:
 		mode = os.O_RDWR | os.O_CREATE | os.O_TRUNC
 	}
-	f, err := os.OpenFile(r.relPath(arg), mode, 0644)
-	if err != nil {
-		// TODO: print to stderr?
-		return nil, err
+	var f io.ReadWriteCloser
+	switch arg {
+	case "/dev/null":
```

Comment:
Paths like `/dev/../dev/null` can be usually fixed with [`filepath.Clean`](https://golang.org/pkg/path/filepath/#Clean). [Example](https://play.golang.org/p/jvdGjn8VXz).

But on Windows I think it may be converted to something like `C:\dev\null`. Will do a try later.